src/users/users.controller.spec.ts

```diff
@@ -1,25 +1,174 @@
 import { Test, TestingModule } from "@nestjs/testing";
 import { UsersController } from "./users.controller";
 import { UsersService } from "./users.service";
+import { CustomRequest } from "src/global/types/CustomRequest";
+import { UserLookupByEmailDto } from "./dto/lookup-user-by-email.dto";
+import { toBeArray } from "jest-extended";
+import { UnauthorizedException } from "@nestjs/common";
+
+expect.extend({ toBeArray });
 
 describe("UsersController", () => {
     let controller: UsersController;
+    let usersService: UsersService;
+    const usersArr = [
+        {
+            id: "55e03658-7fa8-4815-b4d2-942d2eec1dfb",
+            email: "johndoe@example.com",
+            password: "password",
+            roles: ["user"],
+            isVerified: true,
+        },
+        {
+            id: "27fc995b-040d-415b-a27c-5220f5f00c952",
+            email: "janedoe@example.com",
+            password: "password",
+            role: ["admin"],
+            isVerified: false,
+        },
+    ];
+
+    const userOne = usersArr[0];
+    const userOneId = userOne.id;
+
+    const LookupEmailDtoMock = {
+        email: userOne.email,
+    } as UserLookupByEmailDto;
+
+    const mockUsersService = {
+        findAll: jest.fn(),
+        getProfile: jest.fn(),
+        getPrivateUserProfile: jest.fn(),
+        getUserDetailsById: jest.fn(),
+        getUserDetailsByEmail: jest.fn(),
+    };
 
     beforeEach(async () => {
         const module: TestingModule = await Test.createTestingModule({
             controllers: [UsersController],
-            providers: [
-                {
-                    provide: UsersService,
-                    useValue: {},
-                },
-            ],
-        }).compile();
+            providers: [UsersService],
+        })
+            .overrideProvider(UsersService)
+            .useValue(mockUsersService)
+            .compile();
 
         controller = module.get<UsersController>(UsersController);
+        usersService = module.get<UsersService>(UsersService);
     });
 
     it("should be defined", () => {
         expect(controller).toBeDefined();
     });
+
+    describe("findAll", () => {
+        it("findAll service should be defined", async () => {
+            expect(controller.findAll).toBeDefined();
+        });
+
+        it("should return an array of all users", async () => {
+            (usersService.findAll as jest.Mock).mockResolvedValueOnce(usersArr);
+            const result = await controller.findAll();
+            expect(usersService.findAll).toHaveBeenCalled();
+            expect(result).toBeArray();
+            expect(result).toEqual(usersArr);
+        });
+    });
+
+    describe("getProfile", () => {
+        it("getProfile service should be defined", async () => {
+            expect(controller.getProfile).toBeDefined();
+        });
+
+        it("should return unauthorized if user is not logged in", async () => {
```

Comment:
I think we won't need this as we have tested the unauthorized response in e2e. But we can just leave it here.

Reply:
Thank you for reviewing. I could remove the status code tests since they do seem like repeated work.

Reply:
Hey @cherylli, 

is it ok to mark this conversation as resolved?